sdk/core/switcher/entry.S

```diff
@@ -779,3 +779,77 @@ exception_entry_asm:
 #endif
 #endif // CONFIG_NO_SWITCHER_SAFETY
 	cret
+
+// Returns whether the trusted stack has space for N more calls.
+	.section .text, "ax", @progbits
+	.p2align 2
+	.type __Z23trusted_stack_has_spacei,@function
+__Z23trusted_stack_has_spacei:
+	li                 a2, TrustedStackFrame_size
+	mul                a2, a0, a2
+	cspecialr          ca0, mtdc
```

Comment:
```suggestion
    // Load pointer to trusted stack;
    // this will be used for one load and then clobbered, so can't leak.
	cspecialr          ca0, mtdc
```